base/linalg/triangular.jl

```diff
@@ -1827,8 +1827,12 @@ function eigvecs{T}(A::AbstractTriangular{T})
 end
 det{T}(A::UnitUpperTriangular{T}) = one(T)
 det{T}(A::UnitLowerTriangular{T}) = one(T)
+logdet{T}(A::UnitUpperTriangular{T}) = zero(T)
+logdet{T}(A::UnitLowerTriangular{T}) = zero(T)
 det{T}(A::UpperTriangular{T}) = prod(diag(A.data))
 det{T}(A::LowerTriangular{T}) = prod(diag(A.data))
+logdet{T}(A::UpperTriangular{T}) = sum(log(diag(A.data)))
+logdet{T}(A::LowerTriangular{T}) = sum(log(diag(A.data)))
```

Comment:
Should there be a check for positivity of the diagonal elements or do you just want it to fail?  You can get a positive determinant even with negative elements on the diagonal if there are an even number of them.  In that case the logarithm of the determinant exists but this computation will fail for real `T`.

This may seem trivial but, e.g. the QR decomposition of a rectangular `X` often ends up with negative diagonals in `R` whereas the Cholesky decomposition of X'X gives non-negative diagonals.